packages/help-center/src/components/help-center-contact-form.tsx

```diff
@@ -243,24 +243,84 @@ export const HelpCenterContactForm = () => {
 
 	const [ debouncedMessage ] = useDebounce( message || '', 500 );
 
-	const { data: sibylArticles } = useSibylQuery(
-		debouncedMessage,
-		Boolean( supportSite?.jetpack ),
-		Boolean( supportSite?.is_wpcom_atomic )
+	const enableGPTResponse =
+		config.isEnabled( 'help/gpt-response' ) && ! ( params.get( 'disable-gpt' ) === 'true' );
+
+	const showingSearchResults = params.get( 'show-results' ) === 'true';
+	const showingGPTResponse = enableGPTResponse && params.get( 'show-gpt' ) === 'true';
+
+	const redirectToArticle = useCallback(
+		( event, result ) => {
+			event.preventDefault();
+
+			// if result.post_id isn't set then open in a new window
+			if ( ! result.post_id ) {
+				const tracksData = {
+					search_query: debouncedMessage,
+					force_site_id: true,
+					location: 'help-center',
+					result_url: result.link,
+					post_id: result.postId,
+					blog_id: result.blogId,
+				};
+				recordTracksEvent( `calypso_inlinehelp_article_no_postid_redirect`, tracksData );
+				window.open( result.link, '_blank' );
+				return;
+			}
+
+			const searchResult = {
+				...result,
+				title: preventWidows( decodeEntities( result.title ) ),
+				query: debouncedMessage,
+			};
+			const params = new URLSearchParams( {
+				link: result.link,
+				postId: result.post_id,
+				query: debouncedMessage || '',
+				title: preventWidows( decodeEntities( result.title ) ),
+			} );
+
+			if ( result.blog_id ) {
+				params.set( 'blogId', result.blog_id );
+			}
+
+			navigate( `/post/?${ params }`, searchResult );
```

Comment:
Does `searchResult` do anything here? The type info for navigate here doesn't share any properties with `searchResult`